bee-ledger/src/workers/pruning/prune.rs

```diff
@@ -29,172 +31,230 @@ static NUM_PRUNINGS: AtomicUsize = AtomicUsize::new(0);
+
+    // Keep still relevant SEPs.
+    //
+    // Note:
+    // Currently Bee is reliant on the snapshot file generated by Hornet, which stores the confirmation index
+    // of an SEP along with it. It then keeps it long enough to be (pretty) sure the coordinator would reject a
+    // message directly referencing it. In Bee, however, we wanted to try a different approach, which doesn't
+    // trust the Coordinator's tip selection, and stores the highest confirmation index of any of its direct
+    // approvers instead.
+    //
+    // For the first X milestones we keep the initial SEP set (from the snapshot file) around, after that, we keep
+    // only the necessary SEPs (the ones that will be referenced in future prunings).
+    let filter_curr_seps = Instant::now();
+    if NUM_PRUNINGS.fetch_add(1, Ordering::Relaxed) >= KEEP_INITIAL_SNAPSHOT_SEPS {
+        curr_seps.retain(|_, v| **v > *index);
+    }
+    timings.filter_curr_seps = filter_curr_seps.elapsed();
+
+    metrics.kept_seps = curr_seps.len();
+
+    // Create the union of both sets:
+    new_seps.extend(curr_seps);
+
+    let num_next_seps = new_seps.len();
+
+    metrics.next_seps = num_next_seps;
+
+    // Write the new set of SEPs to the storage.
+    let batch_new_seps = Instant::now();
+    for (new_sep, index) in &new_seps {
+        Batch::<SolidEntryPoint, MilestoneIndex>::batch_insert(storage, &mut batch, new_sep, index)
+            .map_err(PruningError::storage)?;
+    }
+    timings.batch_new_seps = batch_new_seps.elapsed();
+
+    // Replace the old set of SEPs with the new one.
+    let replace_seps = Instant::now();
+    tangle.replace_solid_entry_points(new_seps).await;
+    timings.replace_seps = replace_seps.elapsed();
+
+    // Update entry point index
+    tangle.update_entry_point_index(index);
+
+    let batch_milestones = Instant::now();
+    let (num_bytes, milestone_data_metrics) =
+        batch::prune_milestone_data(storage, &mut batch, index, config.receipts().enabled(), by_size)?;
+    timings.batch_milestone_data = batch_milestones.elapsed();
+
+    byte_length += num_bytes;
+    metrics.receipts = milestone_data_metrics.receipts;
+
+    // Add unconfirmed data to the delete batch.
+    let batch_unconfirmed_data = Instant::now();
+    let (num_bytes, unconfirmed_data_metrics) =
+        batch::batch_prunable_unconfirmed_data(storage, &mut batch, index, by_size)?;
+    timings.batch_unconfirmed_data = batch_unconfirmed_data.elapsed();
+
+    byte_length += num_bytes;
+
+    metrics.messages += unconfirmed_data_metrics.prunable_messages;
+    metrics.edges += unconfirmed_data_metrics.prunable_edges;
+    metrics.indexations += unconfirmed_data_metrics.prunable_indexations;
+
+    // Remove old SEPs from the storage.
+    //
+    // **WARNING**: This operation must come before the batch is committed!
+    //
+    // TODO: consider batching deletes rather than using Truncate. Is one faster than the other? Do we care if its
```

Comment:
Is this something we need to answer within this PR?

Reply:
No...it's not important as the code works as it is. However, on lonely rainy boring days one could try out the other approach, and see if it has benefits.